PORTAL/jobs/_utils.py

```diff
@@ -4479,21 +4631,12 @@ def _extract_kwargs(cls, data, optional, filename):
             raise ValueError(f'missing required data (fields: {missing})')
         return (kwargs, extra) if kwargs else (extra, None)
 
-    def refresh(self, resfile):
-        """Reload from the given file."""
-        fresh = self.load(resfile)
-        # This isn't the best way to go, but it's simple.
-        vars(self).clear()
-        self.__init__(**vars(fresh))
-        return fresh
-
-    def as_jsonable(self, *, withextra=False):
+    def as_jsonable(self, *, withextra: bool = False) -> dict:
         fields = self.FIELDS
         if not fields:
             fields = [f for f in vars(self) if not f.startswith('_')]
         elif withextra:
             fields.extend((getattr(self, '_extra', None) or {}).keys())
-        optional = set(self.OPTIONAL or ())
```

Comment:
I probably meant to use this in the loop.  Would you mind leaving it and adding a TODO comment about sorting it out?